src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books where id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(INSERT_QUERY,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        "Creating book failed , no rows affected " + book
+                );
+            }
+            book.setId(getGenerateId(statement));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a new book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Optional<Book> book = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                book = Optional.of(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Can't find a book with id = %d ", id), e
+            );
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> list = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                list.add(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB", e);
+        }
+        return list;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        String.format(
+                                "Book with id = %d wasn't updated, no rows affected", book.getId()
```

Comment:
The exception message is repeated twice. Ensure that the message is clear and concise.